File: php/features.php
<?php
declare(strict_types=1);

// UsptoApiCatalog SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UsptoApiCatalogFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UsptoApiCatalogBaseFeature();
            case "test":
                return new UsptoApiCatalogTestFeature();
            default:
                return new UsptoApiCatalogBaseFeature();
        }
    }
}

File: py/features.py
# UsptoApiCatalog SDK feature factory

from feature.base_feature import UsptoApiCatalogBaseFeature
from feature.test_feature import UsptoApiCatalogTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UsptoApiCatalogBaseFeature(),
        "test": lambda: UsptoApiCatalogTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
